# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1425508465 18000
#      Wed Mar 04 17:34:25 2015 -0500
# Node ID 793230bb81d44e3d74cad8160f6ea762f03c5ffc
# Parent  3b67c46065d40fbfb05f210e1a6fbed77294e974
don't update workspace viewer unless at top level (bug #44156)

* octave-qt-link.cc (octave_qt_link::do_set_workspace): Return
immediately if not at top level.

diff --git a/libgui/src/octave-qt-link.cc b/libgui/src/octave-qt-link.cc
--- a/libgui/src/octave-qt-link.cc
+++ b/libgui/src/octave-qt-link.cc
@@ -372,16 +372,19 @@ octave_qt_link::do_execute_command_in_te
 {
   emit execute_command_in_terminal_signal (QString::fromStdString (command));
 }
 
 void
 octave_qt_link::do_set_workspace (bool top_level,
                                   const std::list<workspace_element>& ws)
 {
+  if (! top_level)
+    return;
+
   QString scopes;
   QStringList symbols;
   QStringList class_names;
   QStringList dimensions;
   QStringList values;
   QIntList complex_flags;
 
   for (std::list<workspace_element>::const_iterator it = ws.begin ();
